feat: Add time since last kill function [skip ci]

chewack0 opened the pull request:
A function that returns the time, in milliseconds, since the player last killed a mob.
Example usage: a bar to track killstreak boons in a lootruns.

chewack0 committed:
feat: Add time since last kill function

ShadowCat117 commented on common/src/main/java/com/wynntils/core/consumers/functions/FunctionManager.java:
```diff
@@ -467,6 +467,7 @@ private void registerAllFunctions() {
         registerFunction(new CombatFunctions.TicksSinceLastSpellFunction());
         registerFunction(new CombatFunctions.TimeSinceLastDamageDealtFunction());
         registerFunction(new CombatFunctions.TotalAreaDamageFunction());
+        registerFunction(new CombatFunctions.TimeSinceLastKillFunction());
```
Shift this up 1 just to keep it alphabetical

ShadowCat117 commented on common/src/main/java/com/wynntils/models/combat/CombatModel.java:
```diff
@@ -68,6 +70,14 @@ public long getLastDamageDealtTimestamp() {
+            return Math.max(lastSelfKillTimestamp, lastSharedKillTimestamp);
+        } else {
+            return lastSelfKillTimestamp;
+        }
```
```suggestion
        return includeShared ? Math.max(lastSelfKillTimestamp, lastSharedKillTimestamp) : lastSelfKillTimestamp;
```

kristofbolyai replied:
Plus, while you are here, please move this method and the one above below the events.

ShadowCat117 reviewed (changes requested):
2 minor things, other than it looks great! Thanks for the contribution :)

chewack0 committed:
refactor: Address review comments

- Moved methods below the events
- Put register in alphabetical order
- Refactored getLastKillTimestamp method

ShadowCat117 reviewed (approved):
Looks good to me!
